virtio-devices/src/balloon.rs

```diff
@@ -299,12 +299,19 @@ impl BalloonEpollHandler {
                         self.release_memory_range(desc_chain.memory(), range_base, range_len)?;
                     }
                     1 => {
-                        Self::advise_memory_range(
-                            desc_chain.memory(),
-                            range_base,
-                            range_len,
-                            libc::MADV_WILLNEED,
-                        )?;
+                        let page_size = get_page_size() as usize;
+                        let mut rbase = align_page_size_down(range_base.0);
+                        let mut rlen = range_len + (range_base.0 - rbase) as usize;
+                        while rlen > 0 {
+                            Self::advise_memory_range(
+                                desc_chain.memory(),
+                                vm_memory::GuestAddress(rbase),
+                                page_size,
+                                libc::MADV_WILLNEED,
+                            )?;
+                            rbase += page_size as u64;
+                            rlen -= page_size;
+                        }
```

Comment:
Maybe we have different understanding of this. I think the range_len can be random size,so the memory range can cover serveral pages not just one or two. Thus, we need count it one by one in a loop. Am I right?

Reply:
No, I believe the `range_len` will always be 4K which will be spanned up-to 2 pages no matter what's the host page size is.

Reply:
Yes, the range_len is always 4K. Also, the range_base aligns with 4K which can guarantee the range locates in one page. So, we can simplify these code.

Reply:
Well. `range_base` aligns with 4K does not guarantee it aligns with the host page size (say it may not align with 2M), e.g. it may span up-to 2 pages.